pkg/mcs/registry/registry.go

```diff
@@ -44,26 +44,18 @@ type RegistrableService interface {
 // It implements the `ServiceRegistry` interface.
 type ServiceRegistry struct {
 	builders map[string]ServiceBuilder
-	services map[string]RegistrableService
```

Comment:
Remove this field otherwise the local integration test will "share" the same builder for the whole cluster, which will cause undefined behavior.